src/core/algorithms/nar/des/des.cpp

```diff
@@ -0,0 +1,95 @@
+#include "des.h"
+
+#include <algorithm>
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "algorithms/nar/value_range.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "model/types/types.h"
+
+namespace algos::des {
+using model::ValueRange;
+
+DES::DES() : NARAlgorithm({}) {
+    RegisterOptions();
+}
+
+void DES::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+
+    DifferentialStrategy default_strategy = DifferentialStrategy::rand1Bin;
+    RegisterOption(Option{&population_size_, kPopulationSize, kDPopulationSize, 100u});
+    RegisterOption(
+            Option{&num_evaluations_, kMaxFitnessEvaluations, kDMaxFitnessEvaluations, 1000u});
+    RegisterOption(Option{&differential_options_.differential_scale, kDifferentialScale,
+                          kDDifferentialScale, 0.5});
+    RegisterOption(Option{&differential_options_.crossover_probability, kCrossoverProbability,
+                          kDCrossoverProbability, 0.9});
+    RegisterOption(Option{&differential_options_.differential_strategy, kDifferentialStrategy,
+                          kDDifferentialStrategy, default_strategy});
+}
+
+void DES::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kPopulationSize, kMaxFitnessEvaluations, kDifferentialScale,
+                          kCrossoverProbability, kDifferentialStrategy});
+}
+
+FeatureDomains DES::FindFeatureDomains(TypedRelation const* typed_relation) {
+    std::vector<std::shared_ptr<ValueRange>> feature_domains;
+    feature_domains.reserve(typed_relation->GetNumColumns());
+    for (size_t i = 0; i < typed_relation->GetNumColumns(); ++i) {
+        auto const& column = typed_relation->GetColumnData(i);
+        feature_domains.push_back(CreateValueRange(column));
+    }
```

Comment:
So sorry, don't know how I missed this, but that is just a 
```suggestion
    std::ranges::transform(typed_relation->GetColumnData(), std::back_inserter(feature_domains), CreateValueRange);
```